### PR TITLE
feat(blocks): connector label

### PR DESCRIPTION
**Pull request by fundon**
## Connector Label Feature

The `label` is along the connector path. 
We follow the [`offset`](https://developer.mozilla.org/en-US/docs/Web/CSS/offset) CSS property design.

The following are the new fields of the connector element.

```ts
export enum ConnectorLabelOffsetAnchor {
  Top = 'top',
  Center = 'center',
  Bottom = 'bottom',
}

export type ConnectorLabelOffsetProps = {
  // [0, 1], `0.5` by default
  distance: number;
  // `center` by default
  anchor?: ConnectorLabelOffsetAnchor;
};

export type ConnectorLabelConstraintsProps = {
  hasMaxWidth: boolean;
  maxWidth: number;
};

export type ConnectorLabelProps = {
  // Label's content
  text?: Y.Text;
  labelEditing?: boolean;
  labelDisplay?: boolean;
  labelXYWH?: XYWH;
  labelOffset?: ConnectorLabelOffsetProps;
  labelStyle?: TextStyleProps;
  labelConstraints?: ConnectorLabelConstraintsProps;
};
```

### Changes

* Adjust the code style of the html template
* Merge some redundant types: `FontFamily`, `FontStyle`, `FontWeight`, `TextAlign`, `StrokeStyle`
* Delete redundant components

* [ ] Add test cases

### CODE REVIEW

**Comment by fundon** [comment]
* **#7106** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7106?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/surface-block/canvas-renderer/element-renderer/text/utils.ts [hunk 1]
```diff
@@ -434,14 +434,26 @@ export function getCursorByCoord(
...
+    fontSize,
+    fontFamily,
+    textMaxWidth,
+  }: {
+    yText: Y.Text;
+    fontStyle: FontStyle;
+    fontWeight: FontWeight;
+    fontSize: number;
+    fontFamily: FontFamily;
+    textMaxWidth: boolean | number;
```
Comment (doodlewind) [inline packages/blocks/src/surface-block/canvas-renderer/element-renderer/text/utils.ts:450]: Should be better using a boolean field for wrap, and another optional number.

File: packages/blocks/src/surface-block/element-model/connector.ts [hunk 2]
```diff
@@ -37,6 +51,21 @@ export enum ConnectorMode {
...
+  labelDisplay?: boolean;
+  labelXYWH?: XYWH;
+  offsetDistance?: number;
+  color?: string;
+  fontFamily?: FontFamily;
+  fontSize?: number;
+  fontStyle?: FontStyle;
+  fontWeight?: FontWeight;
+  textAlign?: TextAlign;
+  textMaxWidth?: boolean | number;
```
Comment (doodlewind) [inline packages/blocks/src/surface-block/element-model/connector.ts:66]: Should be better using a boolean field for wrap, and another optional number.

File: packages/blocks/src/surface-block/element-model/connector.ts [hunk 3]
```diff
@@ -37,6 +51,21 @@ export enum ConnectorMode {
   Curve,
 }
 
+export type ConnectorLabelProps = {
+  lableEditing?: boolean; // local
+  label?: Y.Text;
+  labelDisplay?: boolean;
+  labelXYWH?: XYWH;
+  offsetDistance?: number;
+  color?: string;
```
Comment (doodlewind) [inline packages/blocks/src/surface-block/element-model/connector.ts:60]: Having a `labelStyle` prop object could help avoiding the conflict with stroke style.

**Review by Svaney-ssman — APPROVED** [review]
GOOGOOD

File: packages/blocks/src/root-block/widgets/element-toolbar/change-text-menu.ts [hunk 4]
```diff
@@ -37,19 +42,75 @@ import {
...
+
+function updateFields(
+  service: EdgelessRootService,
+  element: EdgelessCanvasTextElement,
+  props: { [K in keyof TextStyleProps]?: TextStyleProps[K] }
+) {
+  if (element instanceof ConnectorElementModel) {
+    service.updateElement(element.id, {
+      labelStyle: {
+        ...element.labelStyle,
```
Comment (fundon) [inline packages/blocks/src/root-block/widgets/element-toolbar/change-text-menu.ts:105]: @doodlewind @doouding Is there a better way?

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**